crates/sui-types/src/base_types.rs

```diff
@@ -205,6 +205,7 @@ impl From<&AuthorityPublicKeyBytes> for SuiAddress {
         let g_arr = hasher.finalize();
 
         let mut res = [0u8; SUI_ADDRESS_LENGTH];
+        // TODO: should this panic?
         res.copy_from_slice(&AsRef::<[u8]>::as_ref(&g_arr)[..SUI_ADDRESS_LENGTH]);
```

Comment:
@kchalkias wondering your thought on this: we made assumptions like this one in few places, that the sha3_256 length has to be longer than `SUI_ADDRESS_LENGTH` or `OBJECT_ID` length (see few other "TODO: should this panic")

while this is unsafe (imagine we one day changes the length to be much longer, this slice will panic), im not sure how we should handle? returning an Result here and let the caller to handle might be an overkill.